From: Iain Lane <laney@debian.org>
Date: Thu, 3 Oct 2019 14:25:04 +0200
Subject: Make sure to pass --systemd when we're managing the user session
 with systemd

This is a fixup for the "System X11 Default" session; at the minute real
sessions are expected to pass --systemd or --builtin as appopriate or be
happy with the default.

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=941640
Forwarded: not-needed
---
 gnome-session/gnome-session.in | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/gnome-session/gnome-session.in b/gnome-session/gnome-session.in
index ddd1a59..7a73610 100755
--- a/gnome-session/gnome-session.in
+++ b/gnome-session/gnome-session.in
@@ -29,4 +29,8 @@ if [ -n "$REGION" ]; then
   fi
 fi
 
-exec @libexecdir@/gnome-session-binary "$@"
+if [ -d "${XDG_RUNTIME_DIR}/systemd" ]; then
+  exec @libexecdir@/gnome-session-binary --systemd "$@"
+else
+  exec @libexecdir@/gnome-session-binary --builtin "$@"
+fi
